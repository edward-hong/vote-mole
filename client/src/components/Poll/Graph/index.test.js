import React from 'react'

import Graph from '.'
import renderWithRedux from '../../../utils/test/renderWithRedux'

describe('Graph', () => {
	it('returns `null` when there is no `poll` prop', () => {
		const { container } = renderWithRedux(<Graph />)

		expect(container.firstChild).toBeNull()
	})

	it('renders correctly when `poll` prop is passed in', () => {
		const poll = {
			pollQuestion: 'Favourite Language',
			pollOptions: [
				{
					votes: 3,
					option: 'English',
				},
				{
					votes: 6,
					option: 'French',
				},
				{
					votes: 9,
					option: 'German',
				},
				{
					votes: 5,
					option: 'Russian',
				},
			],
		}
		const { container } = renderWithRedux(<Graph poll={poll} />, {
			initialState: { poll: poll },
		})

		expect(container.firstChild).toMatchSnapshot()
	})
})
