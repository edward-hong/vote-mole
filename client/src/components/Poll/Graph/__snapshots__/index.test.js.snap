// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph renders correctly when \`poll\` prop is passed in 1`] = `
<canvas
  class="chartjs-render-monitor"
  height="0"
  style="display: block;"
  width="0"
/>
`;
